Sources/NetworkServiceTestHelper/MockNetworkService.swift

```diff
@@ -57,27 +57,35 @@
             } catch {
                 return .failure(Failure.unknown(error as NSError))
             }
+            let output = next.output
             switch delay {
             case .infinite:
-                return await Task {
-                    try await scheduler.sleep(for: .seconds(.max))
-                    return try next.output.get()
-                }
-                .result.mapToNetworkError()
+
+                return await Task { [scheduler] in
+                    do {
+                        try await scheduler.sleep(for: .seconds(.max))
```

Comment:
Scheduler throwing concurrency warnings.

Reply:
MockNetworkService needs a major refactor to be concurrency safe. That'll have to be a future PR.